doc/api/test.md

```diff
@@ -219,6 +219,64 @@ test('a test that creates asynchronous activity', (t) => {
 });
 ```
 
+## Running tests from the command line
+
+The Node.js test runner can be invoked from the command line by passing the
+[`--test`][] flag:
+
+```bash
+node --test
+```
+
+By default, Node.js will recursively search the current directory for
+JavaScript source files matching a specific naming convention. Matching files
+are executed as test files. More information on the expected test file naming
+convention and behavior can be found in the [test runner execution model][]
+section. If the current directory contains a directory named `test`, Node.js
+will also recursively search that directory for any file with the extensions
+`.js`, `.cjs`, and `.mjs`.
+
+Alternatively, one or more paths can be provided as the final argument(s) to
+the Node.js command, as shown below.
+
+```bash
+node --test test1.js test2.mjs custom_test_dir/
+```
+
+In this example, the test runner will execute the files `test1.js` and
+`test2.mjs`. The test runner will also recursively search the
+`custom_test_dir/` directory for test files to execute.
+
+### Test runner execution model
+
+When searching for test files to execute, the test runner behaves as follows:
+
+* Any files explicitly provided by the user are executed.
+* If the user did not explicitly specify any paths, the current working
+  directory is recursively searched for files as specified in the following
+  steps.
+* If the user did not explicitly specify any paths and the current working
+  directory contains a directory named `test`, the `test` directory is
+  recursively searched for all `.js`, `.cjs`, and `.mjs` files.
+* `node_modules` directories are skipped unless explicitly provided by the
+  user.
+* `.js`, `.cjs`, and `.mjs` files matching the following patterns are
```

Comment:
this still needs updating to cover a top-level `test` or `tests` dir

Reply:
There were docs on that a couple lines up. I've updated the implementation and the docs. I decided it doesn't make sense to treat `$CWD/test` differently from any other directory named `test` - it would be inconsistent for the same directory to load one set of tests if it is `$CWD/test` and a potentially different set of tests if it wasn't immediately under `$CWD`.

Reply:
It would be consistent with what much of the ecosystem does.

Commands are always ran from the root, and it is expected that they do something different if ran from another location.

Reply:
I noticed I didn't mention this in my previous comment - `test` is still supported, it is just handled the same whether it's accessed explicitly or implicitly, and it doesn't have to be exactly `$PWD/test`.